src/matrixrtc/CallMembership.ts

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixEvent, RoomMember } from "../matrix";
+import { deepCompare } from "../utils";
+import { Focus } from "./focus";
+
+type CallScope = "m.room" | "m.user";
+
+// Represents an entry in the memberships section of an m.call.member event as it is on the wire
+export interface CallMembershipData {
+    application?: string;
+    call_id: string;
+    scope: CallScope;
+    device_id: string;
+    created_ts?: number;
+    expires: number;
+    foci_active?: Focus[];
+}
+
+export class CallMembership {
+    public static equal(a: CallMembership, b: CallMembership): boolean {
+        return deepCompare(a.data, b.data);
+    }
+
+    public constructor(private parentEvent: MatrixEvent, private data: CallMembershipData) {
```

Comment:
Could we please use an `opts` object here in case the amount of params grows?

Reply:
For this one I feel like if we're adding more params, something's gone wrong, although in any case it's not part o the library API so we can change it on a whim.